Added Ubuntu 14.04 zfs support

Kovrinic opened the pull request:
## Overview
For Ubuntu 14.04, [ZoL](https://zfsonlinux.org/) page gives the [zfs-native](https://launchpad.net/~zfs-native/+archive/ubuntu/stable) PPA.  It's currently using [zfs-06.5.11](https://github.com/zfsonlinux/zfs/releases/tag/zfs-0.6.5.11) revision.  This poses a few issues with the current [zfs-linux](https://github.com/librenms/librenms-agent/blob/d49fe954dfdeffbeee091051f1f0c515d020f281/snmp/zfs-linux) script:
1. Run Permissions:
    - [zfs-0.7.0](https://github.com/zfsonlinux/zfs/releases/tag/zfs-0.7.0) changed permissions so non-privileged users can run `zpool list`
2. Parsable list flag:
    - [zfs-0.7.0-rc1](https://github.com/zfsonlinux/zfs/releases/tag/zfs-0.7.0-rc1) added the `-p` in https://github.com/zfsonlinux/zfs/commit/2a8b84b747cb27a175aa3a45b8cdb293cde31886
3. Ubuntu 14.04 supports [Python 3.4.0](https://packages.ubuntu.com/trusty/python3) as `python3`:
    - `subprocess.run()` command works with Python 3.5 and above.

### Fix 1
Add `snmp` to the **sudoers** file via `visudo`,
```bash
snmp  ALL=(root) NOPASSWD: /etc/snmp/zfs-linux
```
and added `sudo` to the extend command.
```bash
extend zfs /usr/bin/sudo /etc/snmp/zfs-linux
```
Even though `snmp` is ran as root, it still needs the extra permissions in the sudoers file.

### Fix 2
Catch `-p` error within the [list](https://github.com/Kovrinic/librenms-agent/blob/166d1022f3671815af7d2d2cd6ae42b6d5d7f2d0/snmp/zfs-linux#L112:L120) command, and instead get the parsable values from the [get](https://github.com/Kovrinic/librenms-agent/blob/166d1022f3671815af7d2d2cd6ae42b6d5d7f2d0/snmp/zfs-linux#L135:L145) command.

### Fix 3
The user will need to manually install `python3.5` _(or higher)_ as the system wide `python3` interpreter.  If they wish to preserve the system `python3` interpreter, then can install from source with `make altinstall` and then add full python3.# path to this script's [shebang](https://github.com/Kovrinic/librenms-agent/blob/166d1022f3671815af7d2d2cd6ae42b6d5d7f2d0/snmp/zfs-linux#L1).

Example: my system has `python3.7` installed via `make altinstall`.
```bash
$ which python3.7
/usr/local/bin/python3.7
```
Change [shebang](https://github.com/Kovrinic/librenms-agent/blob/166d1022f3671815af7d2d2cd6ae42b6d5d7f2d0/snmp/zfs-linux#L1) path to
```py
#!/usr/local/bin/python3.7
```

This is excessive detail for installing and using alternative version of Python, but wanted to outline the basics for posterity.

laf commented:
Does this maintain backwards compatibility, i.e no changes needed in the main librenms repo?

Kovrinic commented:
> Does this maintain backwards compatibility, i.e no changes needed in the main librenms repo?

Correct.  The script only adds fallback steps to allow users with older OS and/or ZoL a chance to return the same monitored fields.

To recap: users with older OS and/or ZoL will need to add `snmp` to the **sudoers** file via `visudo`, install python 3.5 (or above) and then potentially edit this file's `shebang` to accommodate the different python version.

---

- Tested with Python `3.5.6`, `3.6.3`, & `3.7.1`
- OS: Ubuntu 14.04.5 LTS x86_64 _(3.13.0-163-generic)_

VVelox commented:
This is actually a common issue here for lots of the extends as snmpd is fairly commonly running as non-root. If you don't mind me suggesting, it is generally better to run sudo the extend line in snmpd.conf and have it call the script.

That way customization is not required for every script, but can be handled via the snmpd config instead.

Kovrinic commented:
@VVelox

Moved `sudo` call to `snmpd.conf` and updated original https://github.com/librenms/librenms-agent/pull/213#issue-234499282 to address the changes in https://github.com/librenms/librenms-agent/pull/213/commits/166d1022f3671815af7d2d2cd6ae42b6d5d7f2d0.

PipoCanaja commented:
Hi @VVelox 
What is the status of this PR ?

SourceDoctor commented:
@Kovrinic @VVelox @PipoCanaja 
can someone verify this PR is ready for merging?
No activity since 4 month ...

Kovrinic commented:
@SourceDoctor, PR works fine on my end _(still using Ubuntu 14.04)_, but can't speak for others.

SourceDoctor commented:
i don't have such old Systems on running, but code looks good to me.
